homeassistant/components/binary_sensor/august.py

```diff
@@ -60,24 +68,55 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     for doorbell in data.doorbells:
         for sensor_type in SENSOR_TYPES:
+            if SENSOR_TYPES[sensor_type][1] == 'door':
+                continue
+            _LOGGER.debug("Adding doorbell sensor class %s for %s",
+                          SENSOR_TYPES[sensor_type][1], doorbell.device_name)
             devices.append(AugustBinarySensor(data, sensor_type, doorbell))
 
+    from august.lock import LockDoorStatus
+    for lock in data.locks:
+        for sensor_type in SENSOR_TYPES:
+            if SENSOR_TYPES[sensor_type][1] != 'door':
+                continue
+
+            state_provider = SENSOR_TYPES[sensor_type][2]
+            if state_provider(data, lock) is LockDoorStatus.UNKNOWN:
+                _LOGGER.debug(
+                    ("Not adding sensor class %s for lock %s "
+                     "as status is unknown"),
+                    SENSOR_TYPES[sensor_type][1], lock.device_name
+                )
+            else:
+                _LOGGER.debug(
+                    "Adding lock sensor class %s for %s",
+                    SENSOR_TYPES[sensor_type][1], lock.device_name
+                )
+                devices.append(AugustBinarySensor(data, sensor_type, lock))
+
     add_entities(devices, True)
 
 
 class AugustBinarySensor(BinarySensorDevice):
     """Representation of an August binary sensor."""
 
-    def __init__(self, data, sensor_type, doorbell):
+    def __init__(self, data, sensor_type, device):
         """Initialize the sensor."""
         self._data = data
         self._sensor_type = sensor_type
-        self._doorbell = doorbell
+        self._device = device
         self._state = None
 
     @property
     def is_on(self):
         """Return true if the binary sensor is on."""
+        from august.lock import LockDoorStatus
+
+        # For door sensor, return true if open or unknown,
```

Comment:
If status is unknown, we should return `False` from the `available` property.

Reply:
I think you misread my comment. You need to return `False` from the **available** property.

Reply:
@balloob ,
Found that snjoetw had provided updates for this component but those were never introduced into HASS. 
Just took his source (as it seemed cleaner and better to me) and added the debug logging I had done; not creating the binary_sensor if the DoorSense sensor does not exist; and fix for issue when battery attribute is not available.